apps/api/src/app/inbox/usecases/update-preferences/update-preferences.usecase.ts

```diff
@@ -171,4 +195,63 @@ export class UpdatePreferences {
       ...(command.level === PreferenceLevelEnum.TEMPLATE && command.workflowId && { _templateId: command.workflowId }),
     };
   }
+
+  private async storePreferences(item: {
+    enabled: boolean;
+    channels: IPreferenceChannels;
+    organizationId: string;
+    subscriberId: string;
+    environmentId: string;
+    templateId?: string;
+  }) {
+    const preferences = {
+      workflow: {
+        defaultValue: item.enabled || PREFERENCE_DEFAULT_VALUE,
+        readOnly: false,
+      },
+      channels: {
+        in_app: {
+          defaultValue: item.channels.in_app || PREFERENCE_DEFAULT_VALUE,
+          readOnly: false,
+        },
+        sms: {
+          defaultValue: item.channels.sms || PREFERENCE_DEFAULT_VALUE,
+          readOnly: false,
+        },
+        email: {
+          defaultValue: item.channels.email || PREFERENCE_DEFAULT_VALUE,
+          readOnly: false,
+        },
+        push: {
+          defaultValue: item.channels.push || PREFERENCE_DEFAULT_VALUE,
+          readOnly: false,
+        },
+        chat: {
+          defaultValue: item.channels.chat || PREFERENCE_DEFAULT_VALUE,
+          readOnly: false,
+        },
+      },
+    };
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
👏 **praise:** Thank for the changes here!‏